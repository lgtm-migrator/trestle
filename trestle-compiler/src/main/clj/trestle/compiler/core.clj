(ns trestle.compiler.core
  (:require [clojure.spec.alpha :as s]
            [cheshire.core :as c]))

(s/def
  ::version
  (s/and pos-int? #(>= 1 %)))

(s/def ::dataset string?)
(s/def ::name string?)
(s/def ::type string?)
(s/def ::spatial boolean?)
(s/def ::identifier boolean?)
(s/def ::temporal string?)

; This isn't working, I want to enforce that we can only have a single qualifier on each fact
; Not sure how to do that
; I'd also like to ensure we have a maximum of 1 ID or spatial value
(s/def ::fact-qualifiers (s/or :spatial ::spatial :identifier ::identifier :temporal ::temporal))

;(s/def ::fact-qualifiers (s/or
;                           (s/keys :req-un [:spatial ::spatial])
;                           (s/keys :req-un [:identifier ::identifier])
;                           (s/keys :req-un [:temporal ::temporal])))

(s/def ::fact-params (s/keys
                       :req-un [::name ::type]
                       :opt-un [::fact-qualifiers]))

(s/def ::fact (s/merge ::fact-qualifiers ::fact-params))

(s/def ::facts (s/coll-of ::fact))

(s/def :dataset/spec
  (s/keys :req-un [::version ::dataset ::facts]))

(defn parse-json [file]
  (c/parse-stream (clojure.java.io/reader file) true))

(s/fdef parse-json
        :args (s/cat :file string?)
        :ret ::dataset)