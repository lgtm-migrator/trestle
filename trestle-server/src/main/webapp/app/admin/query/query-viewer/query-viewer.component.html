Query results: {{queryData?.rows}}
<table class="centered highlight responsive-table" *ngIf="queryData">
    <thead>
    <tr>
        <th *ngFor="let name of queryData.bindingNames">{{name}}</th>
    </tr>
    </thead>
    <tbody>
    <tr *ngFor="let result of queryData.results">
        <td *ngFor="let name of queryData.bindingNames">{{result.resultValues[name]}}</td>
    </tr>
    </tbody>
</table>
