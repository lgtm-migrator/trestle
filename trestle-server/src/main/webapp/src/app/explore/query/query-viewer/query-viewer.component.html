<div class="mat-elevation-z1">
  <h4>Query results: {{queryData.results.length}}</h4>
  <table mat-table [dataSource]="dataSource">

    <ng-container *ngFor="let col of queryData.bindingNames" matColumnDef="{{col}}">
      <th mat-header-cell *matHeaderCellDef>{{col}}</th>
      <td mat-cell *matCellDef="let row">{{row.resultValues[col]}}</td>
    </ng-container>

    <tr mat-header-row *matHeaderRowDef="queryData.bindingNames"></tr>
    <tr mat-row *matRowDef="let row; columns: queryData.bindingNames"></tr>
  </table>
  <mat-paginator [pageSizeOptions]="[5, 10, 20]" showFirstLastButtons></mat-paginator>
</div>
