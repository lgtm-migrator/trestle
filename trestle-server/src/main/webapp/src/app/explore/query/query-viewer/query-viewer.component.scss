.mat-cell {
  padding: 0 10px 0;
}

.mat-table {
  width: 90%;
}
