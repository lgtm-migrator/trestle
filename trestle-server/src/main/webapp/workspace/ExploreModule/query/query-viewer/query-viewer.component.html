<h4>Query results: {{queryData?.rows}}</h4>
<table class="centered highlight responsive-table" *ngIf="queryData">
    <thead>
    <tr>
        <th *ngFor="let name of queryData.bindingNames">{{name}}</th>
    </tr>
    </thead>
    <tbody>
    <tr *ngFor="let result of queryData.results">
        <td *ngFor="let name of queryData.bindingNames">{{result.resultValues[name]}}</td>
    </tr>
    </tbody>
</table>
