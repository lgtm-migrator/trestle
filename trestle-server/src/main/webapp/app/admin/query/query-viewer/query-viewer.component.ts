/**
 * Created by nrobison on 2/28/17.
 */
import {Component, OnChanges, SimpleChanges, Input, Pipe, PipeTransform} from "@angular/core";
import {ITrestleResultSet} from "../query.service";

@Component({
    selector: "query-viewer",
    templateUrl: "./query-viewer.component.html",
    styleUrls: ["./query-viewer.component.css"]
})

export class QueryViewer {
    @Input("data") queryData: ITrestleResultSet;

    constructor() {}

    // ngOnChanges(changes: SimpleChanges): void {
    //     if (!changes["queryData"].isFirstChange()) {
    //         console.debug("Data changed to:", changes["queryData"].currentValue)
    //     }
    // }
}

@Pipe({name: "mapKeys"})
export class MapKeysPipe implements PipeTransform {
    transform(value: any, ...args: any[]): any {
        let returnArray: Array<any> = [];
        Object.keys(value).forEach(key => {
            returnArray.push({
                key: key,
                value: value[key],
            })
        });
        // value.forEach((entryVal: any, entryKey: any) => {
        //     returnArray.push({
        //         key: entryKey,
        //         value: entryVal
        //     });
        // });
        return returnArray;
    }
}