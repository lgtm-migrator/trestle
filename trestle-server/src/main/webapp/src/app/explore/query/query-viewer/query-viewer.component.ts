/**
 * Created by nrobison on 2/28/17.
 */
import {Component, Input} from "@angular/core";
import {ITrestleResultSet} from "../query.service";

@Component({
    selector: "query-viewer",
    templateUrl: "./query-viewer.component.html",
    styleUrls: ["./query-viewer.component.scss"]
})

export class QueryViewerComponent {
    @Input("data") queryData: ITrestleResultSet;

    constructor() {}

    // ngOnChanges(changes: SimpleChanges): void {
    //     if (!changes["queryData"].isFirstChange()) {
    //         console.debug("Data changed to:", changes["queryData"].currentValue)
    //     }
    // }
}
