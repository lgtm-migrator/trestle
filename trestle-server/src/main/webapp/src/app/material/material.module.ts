import {NgModule} from '@angular/core';
import {MatAutocompleteModule} from '@angular/material/autocomplete';
import {MatButtonToggleModule} from '@angular/material/button-toggle';
import {MatExpansionModule} from '@angular/material/expansion';
import {CdkTableModule} from '@angular/cdk/table';
import {MatButtonModule} from '@angular/material/button';
import {MatCardModule} from '@angular/material/card';
import {MatCheckboxModule} from '@angular/material/checkbox';
import {MatChipsModule} from '@angular/material/chips';
import {MatDialogModule} from '@angular/material/dialog';
import {MatFormFieldModule} from '@angular/material/form-field';
import {MatGridListModule} from '@angular/material/grid-list';
import {MatIconModule} from '@angular/material/icon';
import {MatInputModule} from '@angular/material/input';
import {MatListModule} from '@angular/material/list';
import {MatMenuModule} from '@angular/material/menu';
import {MatOptionModule} from '@angular/material/core';
import {MatProgressBarModule} from '@angular/material/progress-bar';
import {MatProgressSpinnerModule} from '@angular/material/progress-spinner';
import {MatRadioModule} from '@angular/material/radio';
import {MatSelectModule} from '@angular/material/select';
import {MatSidenavModule} from '@angular/material/sidenav';
import {MatSliderModule} from '@angular/material/slider';
import {MatSlideToggleModule} from '@angular/material/slide-toggle';
import {MatTableModule} from '@angular/material/table';
import {MatTabsModule} from '@angular/material/tabs';
import {MatToolbarModule} from '@angular/material/toolbar';
import {MatTooltipModule} from '@angular/material/tooltip';
import {MatPaginatorModule} from '@angular/material/paginator';

@NgModule({
  imports: [
    CdkTableModule,
    MatAutocompleteModule,
    MatButtonModule,
    MatButtonToggleModule,
    MatCardModule,
    MatCheckboxModule,
    MatChipsModule,
    MatDialogModule,
    MatExpansionModule,
    MatFormFieldModule,
    MatGridListModule,
    MatIconModule,
    MatInputModule,
    MatListModule,
    MatMenuModule,
    MatOptionModule,
    MatPaginatorModule,
    MatProgressBarModule,
    MatProgressSpinnerModule,
    MatRadioModule,
    MatSelectModule,
    MatSidenavModule,
    MatSliderModule,
    MatSlideToggleModule,
    MatTableModule,
    MatTabsModule,
    MatToolbarModule,
    MatTooltipModule],
  exports: [
    CdkTableModule,
    MatAutocompleteModule,
    MatButtonModule,
    MatButtonToggleModule,
    MatCardModule,
    MatCheckboxModule,
    MatChipsModule,
    MatDialogModule,
    MatExpansionModule,
    MatFormFieldModule,
    MatGridListModule,
    MatIconModule,
    MatInputModule,
    MatListModule,
    MatMenuModule,
    MatOptionModule,
    MatPaginatorModule,
    MatProgressBarModule,
    MatProgressSpinnerModule,
    MatRadioModule,
    MatSelectModule,
    MatSidenavModule,
    MatSliderModule,
    MatSlideToggleModule,
    MatTableModule,
    MatTabsModule,
    MatToolbarModule,
    MatTooltipModule]
})
export class MaterialModule {
}
