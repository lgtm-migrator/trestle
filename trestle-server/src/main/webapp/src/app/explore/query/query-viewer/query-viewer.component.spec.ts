import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import {QueryViewerComponent} from './query-viewer.component';
import {MaterialModule} from '../../../material/material.module';
import {NoopAnimationsModule} from '@angular/platform-browser/animations';

describe('QueryViewerComponent', () => {
  let component: QueryViewerComponent;
  let fixture: ComponentFixture<QueryViewerComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [MaterialModule, NoopAnimationsModule],
      declarations: [ QueryViewerComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(QueryViewerComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
