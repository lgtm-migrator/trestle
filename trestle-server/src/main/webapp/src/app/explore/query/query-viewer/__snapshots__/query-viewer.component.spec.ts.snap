// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryViewerComponent should create 1`] = `QueryViewerComponent {}`;
